File: Assignment2-SSMM/Part2/src/MMExample.java
//Simple example to demonstrate memory management in Java
public class MMExample {
    static class LinkedNode {
        int nodeData;
        LinkedNode nextNode;

        LinkedNode(int data) {
            this.nodeData = data;
        }
    }

    public static void main(String[] args) {
        System.out.println("Allocating new LinkedNode object");
        LinkedNode linkHead = new LinkedNode(20);
        linkHead.nextNode = new LinkedNode(500);

        System.out.println("linkHead data : " + linkHead.nodeData);
        System.out.println("nextNode data: " + linkHead.nextNode.nodeData);

        linkHead = null;
        System.gc();
        System.out.println("Completed!");
    }
}

File: Assignment2-SSMM/Part2/src/MMExample.cpp
//Simple example of memory management in C++ using new and delete operators
#include <iostream>

int main() {
    // Allocate memory on the heap
    std::cout << "Allocating memory on the heap" << std::endl;
    int* my_ptr = new int(799);
    std::cout << "Value: " << *my_ptr << std::endl;
    // Deallocate memory manually
    delete my_ptr;
    my_ptr = nullptr;
    std::cout << "Memory deallocated" << std::endl;
    return 0;
}